subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -713,6 +856,88 @@ static void pe_drs_send_swap_run(void *obj)
 	}
 }
 
+/**
+ * PE_Get_Sink_Cap Entry state
+ * @brief 8.3.3.18.7.1 PE_DR_SRC_Get_Source_Cap State
+ * @brief 8.3.3.18.9.1 PE_DR_SNK_Get_Sink_Cap State
+ */
+void pe_get_sink_cap_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * On entry to the PE_DR_SNK_Get_Sink_Cap state the Policy Engine
+	 * Shall send a Get_Sink_Cap Message and initialize and run the
+	 * SenderResponseTimer.
+	 */
+	pe_send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_DR_SWAP);
+	/* Initialize Submachine */
+	pe->submachine = SM_WAIT_FOR_TX;
+}
+
+/**
+ * @brief PE_Get_Sink_Cap Run state
+ *        NOTE: Sender Response Timer is handled in super state.
+ */
+void pe_get_sink_cap_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	struct usbc_port_data *data = dev->data;
+	struct protocol_layer_rx_t *prl_rx = data->prl_rx;
+	union pd_header header;
+
+	switch (pe->submachine) {
+	case SM_WAIT_FOR_TX:
+		if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+			pe->submachine = SM_WAIT_FOR_RX;
+		}
+		break;
```

Comment:
nit: depending on whether you're using an external or internal TCPC, you might get TX complete notification at the same time as the message.  So in those cases, you can save yourself another state machine run by falling through

Reply:
done